TensorLib/NonEmptyList.lean

```diff
@@ -0,0 +1,25 @@
+namespace TensorLib
+
+inductive NonEmptyList (a: Type) where
```

Comment:
If we have a single constructor, why not a record?

Reply:
Way better.